src/xpath.c

```diff
@@ -5514,8 +5516,15 @@ moveto_node_check(struct lyd_node *node, enum lyxp_node_type root_type, const ch
     }
 
     /* name check */
-    if (strcmp(node_name, "*") && !ly_strequal(node->schema->name, node_name, 1)) {
-        return -1;
+    if (node_name_dict) {
+        if (strcmp(node_name_dict, "*") && !ly_strequal(node->schema->name, node_name_dict, 1)) {
+            return -1;
+        }
+    } else {
+        if (strcmp(node_name, "*") && (node->schema->name[node_name_len] != '\0' ||
+            strncmp(node->schema->name, node_name, node_name_len))) {
```

Comment:
It is not safe in this order because until you call `strncmp` you do not know that `node->schema->name` has at least `node_name_len` characters.

Reply:
Yes, it was still WIP, I'll change here.